Standalone indicators

brockfanning opened the pull request:
This adds support for "standalone" indicators, separate from the SDGs. The most relevant example now would be an indicator page about Covid-19 statistics.

## Testing

As with most new features this depends on two other PRs: [sdg-build](https://github.com/open-sdg/sdg-build/pull/179) and [jekyll-open-sdg-plugins](https://github.com/open-sdg/jekyll-open-sdg-plugins/pull/61). So testing this would involve these changes:

Site repository, _config.yml:

```
 remote_theme: brockfanning/open-sdg@standalone-indicators
```

Site repository, Gemfile:

```
gem 'jekyll-open-sdg-plugins', git: 'https://github.com/brockfanning/jekyll-open-sdg-plugins.git', branch: 'standalone-indicators'
```

Data repository, requirements.txt:

```
git+git://github.com/brockfanning/sdg-build-1@standalone-indicators
```

## Usage

Creating a standalone indicator is almost exactly the same as any other indicator: create a data file and a metadata file. The only difference is that you also add two metadata fields: `standalone` and `permalink`. For example:

```
standalone: true
permalink: my-covid-19-page
```

In fact many typical metadata fields may be unnecessary. A working standalone indicator could consist entirely of this metadata:

```
---
data_non_statistical: false
graph_title: 'My standalone indicator chart title'
graph_type: line
indicator_name: 'My standalone indicator title'
standalone: true
reporting_status: complete
permalink: my-standalone-indicator-path
---
My standalone indicator content
```

## Listing standalone indicators

You may need a way to list some/all of your standalone indicators. A very barebones layout is provided, called "standalone-indicator-list". In your page you would specify exactly which indicators you want to display. Here is an example of a page using this layout:

```
---
title: My list of standalone indicators
layout: standalone-indicators
permalink: /my-standalone-indicator-list
language: en
standalone_indicators:
    - standalone-indicator-1
---
Content for my standalone indicator list
```

Note that the items under `standalone_indicators` correspond to the filenames of the indicator's metadata file, without the extension. So for example if your metadata file is "my-standalone-indicator.md", then you would put:

```
standalone_indicators:
    - my-standalone-indicator
```

### Sorting the list

Note also that the order of the items under `standalone_indicators` will **not** control the order they are listed in. By default they will be listed in alphabetical order according to the metadata filename. You can override this ordering by setting `sort` in the metadata of each standalone indicator.

## Feature branches
Unsorted: http://sdgdev-813006012.eu-west-1.elb.amazonaws.com/standalone-indicators/covid-19-indicators/
Sorted: http://sdgdev-813006012.eu-west-1.elb.amazonaws.com/standalone-indicators-sorted/covid-19-indicators/

LucyGwilliamAdmin commented on _includes/components/indicator/metadata-tabs-default.html:
```diff
@@ -1,9 +1,11 @@
 <li role="presentation" class="nav-item">
   <a class="nav-link" data-toggle="tab" id="tab-global" href="#global" aria-controls="global" role="tab" {% include autotrack.html preset="tab_meta_global" category="Tab change" action="Change metadata view" label="Change to Global Metadata tab" %}>{{ page.t.indicator.global_metadata }}</a>
 </li>
+{% endunlesse %}
```
There's a typo here which is causing an error

brockfanning replied:
Thanks @LucyGwilliamAdmin, just fixed that.

LucyGwilliamAdmin commented:
@brockfanning here's a feature branch:
http://sdgdev-813006012.eu-west-1.elb.amazonaws.com/standalone-indicators/covid-19-indicators/covid-indicator-1/

Everything looks good to me - seems like a good feature to have!

I did notice one thing which I think is probably unrelated, if a chart doesn't have any footnotes, the legend isn't included in the downloaded chart image but I can create an issue for that

brockfanning commented:
@LucyGwilliamAdmin @pixerize I needed to change the order in which these indicators are listed in the standalone-indicators.html layout. This was needed to make it work as expected in combination with other lists, like the previous/next feature.  The way it works now is this:
1. The order of the indicators listed in standalone-indicators.html is determined by the each indicator's filename, by default
2. But you can override this by setting `sort` in that indicator's metadata.

I've updated the description of the PR to document this.

LucyGwilliamAdmin commented:
@brockfanning I seem to be getting this error now:
https://github.com/ONSdigital/sdg-indicators/runs/1576930710?check_suite_focus=true#step:4:124

brockfanning commented:
@LucyGwilliamAdmin I've resolved a conflict in this PR, and I've also added some safety code. The problem was the "covid-indicator-15" [here](https://github.com/ONSdigital/sdg-indicators/blob/standalone-indicators-sorted/_pages/covid-indicators.md) which should have been "covid-indicator-1". The safety code I added will just skip incorrect indicator IDs like this, instead of causing the whole build to fail.

LucyGwilliamAdmin reviewed (approved):
Two feature branches in first comment - one for sorted indicators and one for unsorted

All looks good - also can confirm that the safety code works - indicators that don't exist are skipped

brockfanning commented:
@LucyGwilliamAdmin Hate to ask but could you refresh the feature branches one more time? Both the sdg-build and jekyll-open-sdg-plugins PRs had conflicts and needed to be rebased. There's always the chance I messed something up while resolving the conflicts.

LucyGwilliamAdmin commented:
@brockfanning I've rebuilt both the data branches and site branches but there does seem to be an issue now - the list of standalone indicators isn't showing

brockfanning commented:
@LucyGwilliamAdmin Yep, I did indeed mess something up - it should be fixed now.

LucyGwilliamAdmin commented:
@brockfanning they both seem to be working again now 👍
